packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -63,18 +63,19 @@ const App = (props) => {
     const location = useLocation()
     const authModal = useAuthModal()
     const customer = useCustomer()
+    const appConfig = useAppConfig()
 
-    const site = useSite()
-    const locale = useLocale()
+    const {pathname, search} = useLocation()
+    const site = useMemo(() => {
+        return resolveSiteFromUrl(`${pathname}${search}`)
+    }, [pathname, search])
```

Comment:
What if this calculation of `site` is done within the `SiteProvider`? So that the end users can simply use the Provider and not have to worry about figuring out what the site value is.

Reply:
We also use `site` in the App component to obtain the `site.l10n.defaultCurrency` value.

https://github.com/SalesforceCommerceCloud/pwa-kit/blob/13f6523eff6869240da08903981e333abba4e15a/packages/template-retail-react-app/app/components/_app/index.jsx#L84-L86